wcfsetup/install/files/lib/form/NewPasswordForm.class.php

```diff
@@ -79,10 +78,6 @@ public function readParameters() {
 				$this->throwInvalidLinkException();
 			}
 			
-			(new UserEditor($this->user))->update([
-				'lastLostPasswordRequestTime' => 0,
-				'lostPasswordKey' => null
-			]);
 			WCF::getSession()->register('lostPasswordRequest', $this->user->userID);
 		}
 		else {
```

Comment:
In the `else` case it should be verified that there still is a `lostPasswordKey`. Consider this situation:

a) Browser 'A' opens the form.
b) Browser 'B' opens the form.
c) 'A' submits.
d) 'B' can still submit, but it must not.

The minimal change would be:

1. Store userID + key in the `lostPasswordRequest` session variable (instead of just userID).
2. Verify both the userID + key in the `else` part.